Bottom: 6f25149d3891222cbd706bd0c3e42955f3f735ee
Top:    8c16be6063e5827edac8c1b989a41cc9a328ca91
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-16 14:07:37 -0400

Refresh of efika-stac9766

---

diff --git a/sound/soc/fsl/efika-stac9766.c b/sound/soc/fsl/efika-stac9766.c
index 39b3693..c0ebd7f 100644
--- a/sound/soc/fsl/efika-stac9766.c
+++ b/sound/soc/fsl/efika-stac9766.c
@@ -168,21 +168,7 @@ static const struct snd_kcontrol_new efika_controls[] = {
 		efika_set_spk),
 };
 
-static int efika_stac9766_write(void *control_data, long val, int reg)
-{
-	struct snd_ac97 *ac97 = (struct snd_ac97 *)control_data;
-	ac97->bus->ops->write(ac97, reg, val);
-	return 0;
-}
-
-static int efika_stac9766_read(void *control_data, int reg)
-{
-	struct snd_ac97 *ac97 = (struct snd_ac97 *)control_data;
-
-	return ac97->bus->ops->read(ac97, reg);
-}
-
-static int efika_stac9766_init(struct snd_soc_codec *codec)
+static int efika_init(struct snd_soc_codec *codec)
 {
 
 	/* set up efika codec pins */
@@ -204,7 +190,7 @@ static struct snd_soc_dai_link efika_dai[] = {
 		.stream_name = "AC97 Analog",
 		.cpu_dai = &mpc5200_dai_ac97[MPC5200_AC97_ANALOG],
 		.codec_dai = &stac9766_dai[STAC9766_DAI_AC97_ANALOG],
-		.init = efika_stac9766_init,
+		.init = efika_init,
 	},
 	{
 		.name = "AC97 Digital",
@@ -254,14 +240,12 @@ static int __exit efika_stac9766_remove(struct of_device *op)
 static int efika_stac9766_suspend(struct of_device *op,
 	pm_message_t state)
 {
-	struct snd_soc_codec *codec = platform_get_drvdata(op);
-	return snd_soc_suspend(codec, state);
+	return 0;
 }
 
 static int efika_stac9766_resume(struct of_device *op)
 {
-	struct snd_soc_codec *codec = platform_get_drvdata(op);
-	return snd_soc_resume(codec);
+	return 0;
 }
 
 #else
